uaa_client/authentication.py

```diff
@@ -93,13 +93,21 @@ def get_user_by_email(email):
         of security policies for logins.
         '''
 
+        APPROVED_DOMAINS = getattr(settings, 'UAA_APPROVED_DOMAINS', [])
+
         try:
             return User.objects.get(email__iexact=email)
         except User.DoesNotExist:
-            logger.info(
-                'User with email {} does not exist'.format(email)
-            )
-            return None
+            email_pieces = email.split('@')
+            if email_pieces[1] in APPROVED_DOMAINS:
+                User.objects.create_user(email_pieces[0], email)
```

Comment:
Also getting a bit astronauty here: I wonder if it'd make sense for `ALLOWED_DOMAINS` to contain `*`, which would allow *any* users that pass through cloud.gov auth to be created.  We could always add that kind of functionality later though.